h2o-algos/src/main/java/hex/ensemble/StackedEnsemble.java

```diff
@@ -160,8 +164,15 @@ private Frame prepareLevelOneFrame(String levelOneKey, Key<Model>[] baseModelKey
       baseModels.add(aModel);
       baseModelPredictions.add(predictions);
     }
-
-    return prepareLevelOneFrame(levelOneKey, baseModels.toArray(new Model[0]), baseModelPredictions.toArray(new Frame[0]), actuals);
+    boolean keepLevelOneFrame = isTraining && _parms._keep_levelone_frame;
+    Frame levelOneFrame = prepareLevelOneFrame(levelOneKey, baseModels.toArray(new Model[0]), baseModelPredictions.toArray(new Frame[0]), actuals);
+    if (keepLevelOneFrame) {
+      levelOneFrame.write_lock(_job);
+      levelOneFrame = levelOneFrame.deepCopy(levelOneFrame._key.toString());
```

Comment:
Are we sure that when `isTraining` is true no new Vec are being calculated? Otherwise, we would leak data if we made a defensive copy of Vecs that no-one else has a reference to.

Reply:
@michalkurka  that's why I also tested it with blending (for which there are new Vec predictions computed).
And that's why the copy is done only if, on top of this, user asked to keep the level one frame: as soon as he makes this choice, the Vecs lifecycle depends on the model itself. So we don't leak keys, they're removed with the model.